frontend/src/lib/components/Cards/InsightCard/RecordingsUniversalFiltersDisplay.tsx

```diff
@@ -0,0 +1,266 @@
+import { useValues } from 'kea'
+import React from 'react'
+
+import { IconClock, IconFilter, IconSort } from '@posthog/icons'
+
+import {
+    PROPERTY_FILTER_TYPE_TO_TAXONOMIC_FILTER_GROUP_TYPE,
+    formatPropertyLabel,
+    isAnyPropertyfilter,
+    isCohortPropertyFilter,
+    isPropertyFilterWithOperator,
+} from 'lib/components/PropertyFilters/utils'
+import { PropertyKeyInfo } from 'lib/components/PropertyKeyInfo'
+import { LemonTag } from 'lib/lemon-ui/LemonTag/LemonTag'
+import { Link } from 'lib/lemon-ui/Link'
+import { allOperatorsMapping, capitalizeFirstLetter } from 'lib/utils'
+import { humanFriendlyDurationFilter } from 'scenes/session-recordings/filters/DurationFilter'
+import { urls } from 'scenes/urls'
+
+import { cohortsModel } from '~/models/cohortsModel'
+import { propertyDefinitionsModel } from '~/models/propertyDefinitionsModel'
+import {
+    ActionFilter,
+    DurationType,
+    FilterLogicalOperator,
+    PropertyFilterBaseValue,
+    PropertyGroupFilter,
+    RecordingUniversalFilters,
+    UniversalFiltersGroup,
+    UniversalFiltersGroupValue,
+} from '~/types'
+
+import { DateRangeSummary, InsightDetailSectionDisplay } from './InsightDetails'
+
+function isActionFilter(filter: UniversalFiltersGroupValue): filter is ActionFilter {
+    return (filter as ActionFilter).type !== undefined && 'id' in filter
+}
+
+function isUniversalFiltersGroup(value: UniversalFiltersGroupValue): value is UniversalFiltersGroup {
+    return (value as UniversalFiltersGroup).type !== undefined && (value as UniversalFiltersGroup).values !== undefined
+}
+
+export function CompactUniversalFiltersDisplay({
+    groupFilter,
+    embedded,
+}: {
+    groupFilter: UniversalFiltersGroup | PropertyGroupFilter | null
+    embedded?: boolean
+}): JSX.Element {
+    const { cohortsById } = useValues(cohortsModel)
+    const { formatPropertyValueForDisplay } = useValues(propertyDefinitionsModel)
+
+    // Handle both UniversalFiltersGroup and PropertyGroupFilter
+    const isPropertyGroupFilter =
+        groupFilter &&
+        'values' in groupFilter &&
```

Comment:
Paranoid: Looks a bit fragile, like `values` should always be non-empty list, `PropertyGroupFilter` has a bit different set of arguments compared to `UniversalFiltersGroup` and so.